rfcs/0127-issues-warnings.md

```diff
@@ -0,0 +1,288 @@
+  otherPackage.CVE1234 = "ignore";
+}
+```
+
+Some times, there is the need to set the handler for multiple problems on a package, or for one problem kind across all packages. For this,  `config.problemHandlerDefaultMatchers` exists. It is a list of matchers (currently package name, problem name or problem kind), with an associated handler. If multiple matchers match a single package, the one with the highest level (error > warn > ignore) will be used. Since the user configuration is merged with the default configuration, this means that one can not decrease the handler level below the default value. This is to protect users against accidentally disabling entire classes of notifications. Values from `config.problemHandlers` take precedence.
+
+```nix
+config.problemHandlerDefaultMatchers = [
+  { # Wildcard matcher for everything
+    handler = "warn";
+  }
+  { # Match all security warnings on "hello"
+    package = "hello";
+    kind = "insecure";
+    handler = "error";
+  }
+  { # Match all packages affected by the python2 deprecation
+    name = "python2-eol";
+    handler = "error";
+  }
+  { # Has no effect: the default value is higher
+    kind = "insecure";
+    handler = "ignore";
+  }
+  { # Disallowed: use problemHandlers instead
+    pkgName = "hello";
+    name = "CVE1234";
+    handler = "ignore";
+  }
+]
+```
+
+## Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+### Propagation across transitive dependencies
+
+When a package has a problem that `throw`s, all packages that depend on it will fail to evaluate until that problem is ignored or resolved. Most of the time, this is sufficient.
+
+When the problem requires actions on dependents however, it does not sufficiently inform about all packages that need action. Multiple packages may be annotated with the same problem, in that case it should be given a name and the name should be the same across all instances. Other values like the message or the URL list do not need to be the same and may be adapted sensibly.
+
+### Backwards compatbility, Backporting and stable releases
+
+New problems generally should not be added to stable branches if possible, and also not be backported to them, since it may break evaluation. The same rule applies to other changes to a pacakge's `meta` which may generate a problem and thus lead to evaluation failure too. Scenarios where evaluation failure is a desired goal, for example with unfixable security issues, are obviously exempt from this.
+
+### Removal of packages
+
+The plan is to eventually remove packages with long outstanding problems. The details will be part of future work, but at the very least a package must have a problem whose kind defaults to "error" for at least one full release cycle (so that stable users have sufficient time to be warned and intervene).
+
+If a package needs to be removed for some other reason, the problem kind `removal` should be used instead:
```

Comment:
```suggestion
If a package should to be removed for some reason not mapping to a specific kind, the problem kind `removal` should be used instead:
```

Though I wonder if `removal` should even be a problem kind, there's _always_ a reason why something should be removed. What is the motivation for having this kind?